source/common/network/proxy_protocol.cc

```diff
@@ -61,17 +60,48 @@ void ProxyProtocol::ActiveConnection::onReadWorker() {
     return;
   }
 
-  if (proxy_line.find(PROXY_TCP4) != 0) {
+  // Parse proxy protocol line with format: PROXY TCP4/TCP6 SOURCE_ADDRESS DESTINATION_ADDRESS
+  // SOURCE_PORT DESTINATION_PORT.
+  auto line_parts = StringUtil::splitKeep(proxy_line, " ", true);
```

Comment:
The extra validation is based on @jamessynge uncommited code which added error checking to this function. I think the rationale is that the protocol specifies "exactly one space"